### ISSUE TITLE
Confusing error if return forgotten

### ISSUE THREAD

**Issue by cefn** [rpominov/kefir#124]
I'm using flatMap and I had a chunk of code like...

```
.flatMap( function(topic) {
    Kefir.combine([Kefir.constant(topic), tree.getTopicValueStream(topic)]);
});
```

Which was causing this error to be thrown by Kefir (actual error line was unreported in Webstorm for some reason and had to be manually stepped through to trace it).

    TypeError: Cannot read property '_alive' of undefined

It turned out that it was my error as there should have been a return keyword added to the flatMap callback...

```
.flatMap( function(topic) {
    return Kefir.combine([Kefir.constant(topic), tree.getTopicValueStream(topic)]);
});
```

It would be nice if there were a way for an undefined return value to a flatMap function to throw a meaningful error by catching this exception and reporting that it's probably an undefined return value from a flatMap callback function.

Here's a snapshot of the Kefir call stack leading to the exception thrown when the return value is undefined, and therefore this._active cannot be resolved...

```
 if (this._concurLim === -1 || this._curSources.length < this._concurLim) {
    this._addToCur(toObs(obj));
} else {
```

```
	  _addToCur: function _addToCur(obs) {
	    if (this._active) {

```

**Comment by rpominov** [comment]
Yeah, the overall situation with useful error messages is quite bad now. This and #124 is just two examples, but basically there is almost zero checks for correct input in the code.

I'd like to fix this. Not sure when will be able to work on this though. If anyone want to help, I will very appreciate.
